src/components/organisms/AdminVenueView/components/Spaces/Spaces.tsx

```diff
@@ -0,0 +1,246 @@
+    setSelectedRoom(undefined);
+    setUpdatedRoom({});
+  }, []);
+
+  const updateRoomPosition = useCallback(async (position: Position) => {
+    if (!position) return;
+
+    setUpdatedRoom((room) => ({
+      ...room,
+      x_percent: position.left,
+      y_percent: position.top,
+    }));
+  }, []);
+
+  const updateRoomSize = useCallback(async (size: Dimensions) => {
+    if (!size) return;
+
+    setUpdatedRoom((room) => ({
+      ...room,
+      width_percent: size.width,
+      height_percent: size.height,
+    }));
+  }, []);
+
+  const selectedRoomIndex =
+    venue.rooms?.findIndex((room) => room === selectedRoom) ?? -1;
+
+  return (
+    <div className="Spaces">
+      <div className="Spaces__rooms">
+        {hasSelectedRoom ? (
+          <EditRoomForm
+            room={selectedRoom!}
+            updatedRoom={updatedRoom}
+            roomIndex={selectedRoomIndex}
+            onBackClick={clearSelectedRoom}
+            onDelete={clearSelectedRoom}
+            onEdit={clearSelectedRoom}
+          />
+        ) : (
+          <>
+            <div className="Spaces__background">
+              <div className="Spaces__title">Build your spaces</div>
+            </div>
+            <div>
+              <div
+                className="Spaces__venue-rooms"
+                onClick={() => setShowAdvancedSettings(!showAdvancedSettings)}
+              >
+                <div>Map background</div>
+                <FontAwesomeIcon
+                  icon={!showAdvancedSettings ? faCaretRight : faCaretDown}
+                />{" "}
+              </div>
+              {showAdvancedSettings && (
+                <BackgroundSelect venueName={venue.name} mapBackground={""} />
+              )}
+            </div>
+
+            <div>
+              <div
+                className="Spaces__venue-rooms"
+                onClick={() => setShowRooms(!showRooms)}
+              >
+                <div>{numberOfRooms} Rooms</div>
+                <FontAwesomeIcon
+                  icon={!showRooms ? faCaretRight : faCaretDown}
+                />
+              </div>
+
+              {showRooms && (
+                <div>
+                  {venue.rooms?.map((room, index) => {
+                    return (
+                      <div
+                        key={`${index}-${room.title}`}
+                        className="Spaces__venue-room"
+                        onClick={() => setSelectedRoom(room)}
+                      >
+                        <div
+                          className="Spaces__venue-room-logo"
+                          style={{ backgroundImage: `url(${room.image_url})` }}
+                        />
+                        <div className="Spaces__venue-room-title">
+                          {room.title}
+                        </div>
+                      </div>
+                    );
+                  })}
+                </div>
+              )}
+            </div>
```

Comment:
useMemo

Reply:
Fixed in: https://github.com/sparkletown/sparkle/pull/1583/commits/f8fa0c67c34ac626919737b4bbd112b4bd715553